pkg/plugins/policies/matchers/egress.go

```diff
@@ -0,0 +1,190 @@
+	matchedPolicies := []core_model.Resource{}
+
+	for _, policy := range rl.GetItems() {
+		spec := policy.GetSpec().(core_xds.Policy)
+		if !externalServiceSelectedByTargetRef(spec.GetTargetRef(), es) {
+			continue
+		}
+		matchedPolicies = append(matchedPolicies, policy)
+	}
+
+	sort.Sort(ByTargetRef(matchedPolicies))
+
+	return core_xds.BuildFromRules(map[core_xds.InboundListener][]core_model.Resource{
+		{}: matchedPolicies, // egress always has only 1 listener, so we can use empty key
+	})
+}
+
+// It's not natural for zone egress to have 'to' policies. It doesn't make sense to target
+// external service in the top-level targetRef and specify 'to' array (simply because we don't
+// have access to the external service outbounds). But there are situations when we're
+// targeting external service in the 'to' array, and we need to make adjustments on the Egress, i.e:
+//
+// type: MeshLoadBalancingStrategy
+// spec:
+//
+//	targetRef:
+//	  kind: Mesh
+//	to:
+//	  - targetRef:
+//	      kind: MeshService
+//	      name: external-service-1
+//	    default:
+//	      localityAwareness:
+//	        disabled: true
+//
+// In this case we need to apply the policy to the Egress. The problem is that Egress is
+// a single point for multiple clients. This means we have to specify different configurations
+// for different clients. In order to easily get a list of rules for 'to' policy on Egress
+// we have to convert it to 'from' policy, i.e. the policy above will be converted to artificially
+// created policy:
+//
+// spec:
+//
+//	targetRef:
+//	  kind: MeshService
+//	  name: external-service-1
+//	from:
+//	  - targetRef:
+//	      kind: Mesh
+//	    default:
+//	      localityAwareness:
+//	        disabled: true
+//
+// that's why processToRules() method produces FromRules for the Egress.
+func processToRules(
+	es *core_mesh.ExternalServiceResource,
+	rl core_model.ResourceList,
+) (core_xds.FromRules, error) {
+	matchedPolicies := []core_model.Resource{}
+
+	for _, policy := range rl.GetItems() {
+		spec := policy.GetSpec().(core_xds.Policy)
+
+		to, ok := spec.(core_xds.PolicyWithToList)
+		if !ok {
+			return core_xds.FromRules{}, nil
+		}
+
+		for _, item := range to.GetToList() {
+			if externalServiceSelectedByTargetRef(item.GetTargetRef(), es) {
+				matchedPolicies = append(matchedPolicies, policy)
+			}
+		}
+	}
+
+	sort.Sort(ByTargetRef(matchedPolicies))
+
+	toList := []core_xds.PolicyItemWithMeta{}
+	for _, policy := range matchedPolicies {
+		for _, item := range policy.GetSpec().(core_xds.PolicyWithToList).GetToList() {
+			if !externalServiceSelectedByTargetRef(item.GetTargetRef(), es) {
+				continue
+			}
+			// convert 'to' policyItem to 'from' policyItem
+			artificial := &artificialPolicyItem{
+				conf:      item.GetDefault(),
+				targetRef: policy.GetSpec().(core_xds.Policy).GetTargetRef(),
+			}
+			toList = append(toList, core_xds.BuildPolicyItemsWithMeta([]core_xds.PolicyItem{artificial},
+				policy.GetMeta())...)
```

Comment:
Is this going to make things weird with the inspect API?

Reply:
Not really, pieces of configs correspond to the actual existing policies. So they're correct, but maybe we should be careful with GUI